src/main/java/minegame159/meteorclient/modules/player/DeathPosition.java

```diff
@@ -55,27 +61,32 @@ public DeathPosition() {
     @EventHandler
     private void onTookDamage(TookDamageEvent event) {
         if (mc.player == null) return;
+        
+        if (event.entity.getUuid() != null && event.entity.getUuid().equals(mc.player.getUuid()) && event.entity.getHealth() >= 0) {
+            damagedplayerX = mc.player.getX();
+            damagedplayerY = mc.player.getY();
+            damagedplayerZ = mc.player.getZ(); }
 
         if (event.entity.getUuid() != null && event.entity.getUuid().equals(mc.player.getUuid()) && event.entity.getHealth() <= 0) {
-            deathPos.put("x", mc.player.getX());
-            deathPos.put("z", mc.player.getZ());
-            label.setText(String.format("Latest death: %.1f, %.1f, %.1f", mc.player.getX(), mc.player.getY(), mc.player.getZ()));
+            deathPos.put("x", damagedplayerX);
+            deathPos.put("z", damagedplayerZ);
+            label.setText(String.format("Latest death: %.1f, %.1f, %.1f", damagedplayerX, damagedplayerY, damagedplayerZ));
 
             String time = dateFormat.format(new Date());
-            //ChatUtils.moduleInfo(this, "Died at (highlight)%.0f(default), (highlight)%.0f(default), (highlight)%.0f (default)on (highlight)%s(default).", mc.player.getX(), mc.player.getY(), mc.player.getZ(), time);
-            BaseText msg = new LiteralText("Died at ");
-            msg.append(ChatUtils.formatCoords(mc.player.getPos()));
-            msg.append(".");
-            ChatUtils.moduleInfo(this,msg);
+            ChatUtils.moduleInfo(this, "Died at (highlight)%.0f(default), (highlight)%.0f(default), (highlight)%.0f (default)on (highlight)%s(default).", damagedplayerX, damagedplayerY, damagedplayerZ, time);
+            //BaseText msg = new LiteralText("Died at ");
+            //msg.append(ChatUtils.formatCoords(mc.player.getPos()));
```

Comment:
I don't know what the developers think, but i'd use `msg.append(ChatUtils.formatCoords(new Vec3d(damagedplayerX, damagedplayerY,damagedplayerZ ))))`. The `formatCoords` function allows coords to be clickable in chat (makes baritone path to those coords).

Reply:
yeah thats a good idea. sorry for being so bad at this.